BranchMetrics/android-branch-deep-linking
# SDK-28 -- Android project has warnings on first import

## Reference
SDK-28 -- Android project has warnings on first import

## Description
Cloning into https://github.com/BranchMetrics/android-branch-deep-linking.git AndroidStudio is giving a dozen or so warnings, that mostly relate to outdated build tools.

Fixing these and other compiler warnings

## Testing Instructions
* Sample app should continue to function.
* Unit tests should all pass

The way that I tested this was to 
1. clone the project into one directory, and import the master branch into Android Studio -- noting the warnings and errors.
2. clone into an entirely new directory, switch to the `SDK-28/cleanup_warnings` branch, and note that there are no more warnings.

Opening each of the Java files there should no longer be any compiler warnings either.

## Risk Assessment [`LOW`]
The tools to compile the SDK and TestBed are both updated, however the code changes are only in the TestBed Sample app.


## Reviewer Checklist (To be checked off by the reviewer only)

- Correctness & Style
